block/block_cache.go

```diff
@@ -0,0 +1,45 @@
+package block
+
+import (
+	"sync"
+
+	"github.com/dymensionxyz/dymint/types"
+)
+
+type Cache struct {
+	cache map[uint64]CachedBlock
+	sync.Mutex
+}
+
+func (m *Cache) AddBlockToCache(h uint64, b *types.Block, c *types.Commit) {
+	m.Lock()
+	defer m.Unlock()
+	m.cache[h] = CachedBlock{Block: b, Commit: c}
+	types.BlockCacheSizeGauge.Set(float64(len(m.cache)))
+}
+
+func (m *Cache) DeleteBlockFromCache(h uint64) {
+	m.Lock()
+	defer m.Unlock()
+	delete(m.cache, h)
+	size := len(m.cache)
+	types.BlockCacheSizeGauge.Set(float64(size))
+}
+
+func (m *Cache) GetBlockFromCache(h uint64) (CachedBlock, bool) {
+	m.Lock()
+	defer m.Unlock()
+	ret, found := m.cache[h]
+	return ret, found
+}
+
+func (m *Cache) HasBlockInCache(h uint64) bool {
+	_, found := m.GetBlockFromCache(h)
+	return found
+}
```

Comment:
Could you rename the cache to BlockCache and the methods to Add,Delete,Get,Has?

Reply:
the package is called `block` already